airflow/models/dag.py

```diff
@@ -2120,6 +2120,13 @@ def filter_task_group(group, parent_group):
     def has_task(self, task_id: str):
         return task_id in self.task_dict
 
+    def has_task_group(self, task_group_id: str):
+        return task_group_id in self.task_group_dict
+
+    @property
+    def task_group_dict(self):
+        return {k: v for k, v in self._task_group.get_task_group_dict().items() if k is not None}
```

Comment:
I’m kind of wondering if we really need to exclude `None` (the root group) here. It doesn’t really change how everything behaves.

Also this is likely better to be a `cached_property`.